static/scripts/rewards/render-transaction/render-ens-name.ts

```diff
@@ -1,9 +1,21 @@
 import { ensLookup } from "../cirip/ens-lookup";
 import { app } from "./index";
 
-export async function renderEnsName({ element, address, tokenView = false }: { element: Element; address: string; tokenView?: boolean }): Promise<void> {
-  // const provider = new ethers.providers.Web3Provider(window.ethereum);
-  // const ens = await provider.lookupAddress(address);
+type EnsParams =
+  | {
+      element: Element;
+      address: string;
+      tokenAddress: string;
+      tokenView: true;
+    }
+  | {
+      element: Element;
+      address: string;
+      tokenAddress?: undefined;
+      tokenView?: false;
+    };
```

Comment:
```suggestion
type EnsParams = {
      element: HTMLElement;
      address: string;
      tokenAddress: string;
      tokenView: boolean;
    };
```

Reply:
it's supposed to be a discriminated union that forces tokenAddress to be set if tokenView is true